tests/sentry/incidents/test_action_handlers.py

```diff
@@ -302,3 +301,71 @@ class MsTeamsActionHandlerFireTest(MsTeamsActionHandlerBaseTest, TestCase):
     def test(self):
         alert_rule = self.create_alert_rule()
         self.run_test(self.create_incident(status=2, alert_rule=alert_rule), "fire")
+
+
+@freeze_time()
+class PagerDutyActionHandlerBaseTest(object):
+    @responses.activate
+    def run_test(self, incident, method):
+        from sentry.integrations.pagerduty.utils import build_incident_attachment
+
+        SERVICES = [
+            {
+                "type": "service",
+                "integration_key": "PND4F9",
+                "service_id": "123",
+                "service_name": "hellboi",
+            }
+        ]
+        integration = Integration.objects.create(
+            provider="pagerduty",
+            name="Example PagerDuty",
+            external_id="example-pagerduty",
+            metadata={"services": SERVICES},
+        )
+        integration.add_organization(self.organization, self.user)
+
+        service = PagerDutyService.objects.create(
+            service_name=SERVICES[0]["service_name"],
+            integration_key=SERVICES[0]["integration_key"],
+            organization_integration=integration.organizationintegration_set.first(),
+        )
+
+        action = self.create_alert_rule_trigger_action(
+            target_identifier=service.id,
+            type=AlertRuleTriggerAction.Type.PAGERDUTY,
+            target_type=AlertRuleTriggerAction.TargetType.SPECIFIC,
+            integration=integration,
+        )
+
+        responses.add(
+            method=responses.POST,
+            url="https://events.pagerduty.com/v2/enqueue/",
+            body={},
+            status=202,
+            content_type="application/json",
+        )
+        handler = PagerDutyActionHandler(action, incident, self.project)
+        metric_value = 1000
+        with self.tasks():
+            getattr(handler, method)(metric_value)
+        data = responses.calls[0].request.body
+        assert json.loads(data) == build_incident_attachment(
```

Comment:
@ceorourke Asserting that the payload is the same as the payload we generate with `build_incident_attachment` isn't going to help unless there are also tests for `build_incident_attachment`. Otherwise, a bug with `build_incident_attachment` won't get caught.